game/src/game_core/player.rs

```diff
@@ -27,6 +33,114 @@ impl Default for Player {
+                self.last_damage = 0;
+                let mut popup =
+                    Popup::new(Color::from_rgb(52, 235, 52), WARNINGS[4].to_string(), 5);
+                sender.send(StackCommand::Popup(popup)).unwrap();
+            }
+            // If player takes damage, increase last damage point
+            (change_life, _, _) if change_life < 0 => self.last_damage = 0,
+            // Else, increase last damage point
+            _ => self.last_damage += 1,
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::backend::screen::Screen;
+    use std::sync::mpsc::channel;
+
+    #[test]
+    fn test_case_one_life_regeneration() {
+        let mut gamestate = gamestate::GameState::default();
+        let mut channel = channel();
+        gamestate.set_sender(channel.0);
+        let mut player = Player::default();
+        player.resources.life = u16::MAX;
+        player.resources_change.life = 5;
+        player.last_damage = 1000;
+        player.life_regeneration(gamestate.screen_sender.as_ref().unwrap().clone());
+        assert_eq!(player.resources_change.life, 0);
+        assert_eq!(player.last_damage, 0);
+    }
+
+    #[test]
+    fn test_case_two_life_regeneration() {
+        let mut gamestate = gamestate::GameState::default();
+        let mut channel = channel();
+        gamestate.set_sender(channel.0);
+        let mut player = Player::default();
+        player.resources.life = 1000;
+        player.resources_change.life = 5;
+        player.last_damage = 1000;
+        player.life_regeneration(gamestate.screen_sender.as_ref().unwrap().clone());
+        assert_eq!(player.last_damage, 0);
+    }
+
+    #[test]
+    fn test_case_three_life_regeneration() {
+        let mut gamestate = gamestate::GameState::default();
+        let mut channel = channel();
+        gamestate.set_sender(channel.0);
+        let mut player = Player::default();
+        player.resources.life = 1000;
+        player.resources_change.life = 0;
+        player.last_damage = 900;
+        player.life_regeneration(gamestate.screen_sender.as_ref().unwrap().clone());
+        assert_eq!(player.resources_change.life, 5);
+        assert_eq!(player.last_damage, 0);
+    }
+
+    #[test]
+    fn test_case_four_life_regeneration() {
+        let mut gamestate = gamestate::GameState::default();
+        let mut channel = channel();
+        gamestate.set_sender(channel.0);
+        let mut player = Player::default();
+        player.last_damage = 500;
+        player.resources_change.life = 0;
+        player.life_regeneration(gamestate.screen_sender.as_ref().unwrap().clone());
+        assert_eq!(player.resources_change.life, 0);
+        assert_eq!(player.last_damage, 501);
+    }
+
+    #[test]
+    fn test_case_five_life_regeneration() {
+        let mut gamestate = gamestate::GameState::default();
+        let mut channel = channel();
+        gamestate.set_sender(channel.0);
```

Comment:
Wenn wir das in drei tests schreiben könnten wir das fast als Funktion in den Tests schreiben können(nicht als Test)

Reply:
ich mein wir machen das fünf mal xD